### PR TITLE
docs: add clarification regarding development in older browsers in Deployment guide

### PR DESCRIPTION
**Pull request by santoshyadavdev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #32199 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/deployment.md [hunk 1]
```diff
@@ -596,7 +596,7 @@ In `angular.json` add two new configuration sections under the `build` and `serv
 
 </code-example>
 
-You can then run the serve with this configuration.
+You can then run the serve with this configuration. Make sure to replace app with the actual app name in `"browserTarget":"app:build:es5"`.
```
Comment (kapunahelewong) [inline aio/content/guide/deployment.md]: When you say "to replace app with...", what line are you referring to? It looks like you're referring to the command that comes immediately afterward: `ng serve --configuration es5` but that doesn't have "app".

File: aio/content/guide/deployment.md [hunk 2]
```diff
@@ -668,10 +668,11 @@ Create an [ES5 serve configuration](guide/deployment#configuring-serve-for-es5)
 
 </code-example>
 
-You can then run the e2e's with this configuration
+You can then run the e2e's with this configuration. Make sure to replace app with the actual app name in `"devServerTarget": "app:serve:es5"`.
```
Comment (kapunahelewong) [inline aio/content/guide/deployment.md]: Same question here about "app". 

Additionally, since you're in there, could you change "e2e's" to "e2e tests" for clarity?

**Review by kapunahelewong — CHANGES_REQUESTED** [review]
Thanks for this, @santoshyadav198613. Just a couple of questions.

File: aio/content/guide/deployment.md (hunk 1 above)
Comment (santoshyadav198613) [reply aio/content/guide/deployment.md]: I am referring to browserTarget":"app:build:es5

Comment (kapunahelewong) [reply aio/content/guide/deployment.md]: Right, (as per your change), however, the line that follows is the cli command `ng serve --configuration es5`. Are you saying to change `es5` to whatever the name of your app is? Newcomers might not realize that when you say "app" that that's what you're referring to. Since `browserTarget` is only present on the page in that code sample, and not explained anywhere, it warrants an explanation as to what it means for the newcomer. Your change is good, and an explanation would make it even better.

File: aio/content/guide/deployment.md (hunk 2 above)
Comment (santoshyadav198613) [reply aio/content/guide/deployment.md]: Will make the changes .

File: aio/content/guide/deployment.md (hunk 1 above)
Comment (santoshyadav198613) [reply aio/content/guide/deployment.md]: I think we need to add some explanation for what es5 , will make the changes.

File: aio/content/guide/deployment.md (hunk 2 above)
Comment (kapunahelewong) [reply aio/content/guide/deployment.md]: Please tag me when you're ready. Thanks, @santoshyadav198613!

**Comment by santoshyadav198613** [comment]
HI @kapunahelewong ,

Please take a look again,

**Review by kapunahelewong — CHANGES_REQUESTED** [review]
Thanks for this, @santoshyadav198613. Just a few content change requests for you.

File: aio/content/guide/deployment.md [hunk 3]
```diff
@@ -668,10 +668,16 @@ Create an [ES5 serve configuration](guide/deployment#configuring-serve-for-es5)
 
 </code-example>
 
-You can then run the e2e's with this configuration
+You can then run the e2e with this configuration. Make sure to replace `app` with the actual `app` name in for example if your app name is `myAngularApp` the command will be  `"browserTarget":"myAngularApp:serve:es5"`.
```
[the remark this answers is not in the document]
Comment (santoshyadav198613) [reply aio/content/guide/deployment.md]: done

File: aio/content/guide/deployment.md [hunk 4]
```diff
@@ -596,7 +596,7 @@ In `angular.json` add two new configuration sections under the `build` and `serv
 
 </code-example>
 
-You can then run the serve with this configuration.
+You can then run the `ng serve` command with this configuration. Make sure to replace `app` with the actual `app` name in for example if your app name is `myAngularApp` the config will be  `"browserTarget":"myAngularApp:build:es5"`.
```
Comment (gkalpak) [inline aio/content/guide/deployment.md]: > Make sure to replace \`app\` with the actual \`app\` name in for example if your app name is \`myAngularApp\` the config will be  \`"browserTarget":"myAngularApp:build:es5"\`

There are some typos in this sentence. I would rephrase to something like:

> Make sure to replace \`app\` (in \`"app:build:es5"\`) with the actual name of the app, as it appears under \`projects\` in \`angular.json\`. For example, if your app name is \`myAngularApp\` the config will become \`"browserTarget": "myAngularApp:build:es5"\`.

##
Also, to make it more explicit, I would consider changing the code snippet to `"browserTarget": "<app-name>:build:es5"` (and update the wording accordingly).

File: aio/content/guide/deployment.md [hunk 5]
```diff
@@ -668,10 +668,16 @@ Create an [ES5 serve configuration](guide/deployment#configuring-serve-for-es5)
```
Comment (gkalpak) [inline aio/content/guide/deployment.md]: Not directly related to this PR, but (while you're at it) the code snippet above has some issues:

- `"test"` should be `"e2e"`
- The `"production"` configuration section is incorrectly indented.

File: aio/content/guide/deployment.md [hunk 6]
```diff
@@ -668,10 +668,16 @@ Create an [ES5 serve configuration](guide/deployment#configuring-serve-for-es5)
 
 </code-example>
 
-You can then run the e2e's with this configuration
+You can then run the e2e test with this configuration. Make sure to replace `app` with the actual `app` name in for example if your app name is `myAngularApp` the config will be  `"browserTarget":"myAngularApp:serve:es5"`.
```
Comment (gkalpak) [inline aio/content/guide/deployment.md]: _the e2e test_ --> _the \`ng e2e\` command_ or _the e2e test**s**_

Also, my comment from `ng serve` about rephrasing the second sentence applies here too.

File: aio/content/guide/deployment.md [hunk 7]
```diff
@@ -668,10 +668,16 @@ Create an [ES5 serve configuration](guide/deployment#configuring-serve-for-es5)
...
 
 <code-example language="none" class="code-shell">
 
 ng e2e --configuration es5
 
 </code-example>
+
+<div class="alert is-helpful">
+
+In both the configuration `es5` is just a configuration name, you can name it whatever you want. This is not specific to to es5 build, the important thing is the `"./tsconfig-es5.app.json"` file, where we have defined target to `es5`.
```
Comment (gkalpak) [inline aio/content/guide/deployment.md]: TBH, I find that this alert adds little value and might end up being more confusing than helpful.
People that need to name their configurations differently probably will already know enough to understand that `es5` is just an arbitrary name. People that don't know enough to undr=erstand this on their own are better off sticking to the above instructions anyway.

**Review by gkalpak — CHANGES_REQUESTED** [review]
Great addition, @santoshyadav198613, thx :+1:
I left a couple of suggestions. Could you also make the commit message a little more decriptive; e.g.:

> docs: add clarification regarding development in older browsers in Deployment guide

File: aio/content/guide/deployment.md (hunk 5 above)
Comment (santoshyadav198613) [reply aio/content/guide/deployment.md]: Done.

File: aio/content/guide/deployment.md (hunk 4 above)
Comment (santoshyadav198613) [reply aio/content/guide/deployment.md]: Done.

File: aio/content/guide/deployment.md (hunk 7 above)
Comment (santoshyadav198613) [reply aio/content/guide/deployment.md]: Removed.

File: aio/content/guide/deployment.md [hunk 8]
```diff
@@ -589,14 +589,14 @@ In `angular.json` add two new configuration sections under the `build` and `serv
...
-      "browserTarget": "app:build:es5"
+      "browserTarget": "<app-name>:build:es5"
     }
   }
 },
 
 </code-example>
 
-You can then run the serve with this configuration.
+You can then run the `ng serve` command with this configuration. Make sure to replace `app` (in `"app:build:es5"`) with the actual name of the app, as it appears under `projects` in `angular.json`. For example, if your app name is `myAngularApp` the config will become `"browserTarget": "myAngularApp:build:es5"`.
```
Comment (gkalpak) [inline aio/content/guide/deployment.md]: Now that the code snippet has changed, this should be:

> [...] replace \`&lt;app-name&gt;\` (in \`"&lt;app-name&gt;:build:es5"\`) with [...]

File: aio/content/guide/deployment.md [hunk 9]
```diff
@@ -651,15 +651,15 @@ Create an [ES5 serve configuration](guide/deployment#configuring-serve-for-es5)
...
   },
   "configurations": {
-	"production": {
-		...
-	},
+	  "production": {
+		  ...
+	  },
     "es5": {
       "devServerTarget": "app:serve:es5"
```
Comment (gkalpak) [inline aio/content/guide/deployment.md]: Can you update this to match the `serve` config:

```diff
-app:serve:es5
+<app-name>:serve:es5
```

File: aio/content/guide/deployment.md [hunk 10]
```diff
@@ -668,7 +668,7 @@ Create an [ES5 serve configuration](guide/deployment#configuring-serve-for-es5)
 
 </code-example>
 
-You can then run the e2e's with this configuration
+You can then run the `ng e2e` command or e2e tests with this configuration. Make sure to replace `app` (in `"app:serve:es5"`) with the actual name of the app, as it appears under `projects` in `angular.json`. For example if your app name is `myAngularApp` the config will be  `"browserTarget":"myAngularApp:serve:es5"`.
```
Comment (gkalpak) [inline aio/content/guide/deployment.md]: Sorry, I meant to use either "run the \`ng e2e\` command" _or_ "run the e2e tests" (not both).

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: aio/content/guide/deployment.md (hunk 10 above)
Comment (santoshyadav198613) [reply aio/content/guide/deployment.md]: Done.

File: aio/content/guide/deployment.md (hunk 9 above)
Comment (santoshyadav198613) [reply aio/content/guide/deployment.md]: Done.

File: aio/content/guide/deployment.md (hunk 8 above)
Comment (santoshyadav198613) [reply aio/content/guide/deployment.md]: Done.

File: aio/content/guide/deployment.md [hunk 11]
```diff
@@ -651,24 +651,24 @@ Create an [ES5 serve configuration](guide/deployment#configuring-serve-for-es5)
...
-      "devServerTarget": "app:serve:es5"
+      "devServerTarget": "<app-name>:serve:es5"
     }
   }
 },
 
 </code-example>
 
-You can then run the e2e's with this configuration
+You can then run the `ng e2e` command with this configuration. Make sure to replace `<app-name>` (in `"<app-name>:serve:es5"`) with the actual name of the app, as it appears under `projects` in `angular.json`. For example if your app name is `myAngularApp` the config will be  `"browserTarget":"myAngularApp:serve:es5"`.
```
[the remark this answers is not in the document]
Comment (santoshyadav198613) [reply aio/content/guide/deployment.md]: Thanks for pointing that out, done.

File: aio/content/guide/deployment.md [hunk 12]
```diff
@@ -651,24 +651,24 @@ Create an [ES5 serve configuration](guide/deployment#configuring-serve-for-es5)
...
-      "devServerTarget": "app:serve:es5"
+      "devServerTarget": "<app-name>:serve:es5"
     }
   }
 },
 
 </code-example>
 
-You can then run the e2e's with this configuration
+You can then run the `ng e2e` command with this configuration. Make sure to replace `<app-name>` (in `"<app-name>:serve:es5"`) with the actual name of the app, as it appears under `projects` in `angular.json`. For example, if your app name is `myAngularApp` the config will become `"browserTarget":"myAngularApp:serve:es5"`.
```
Comment (gkalpak) [inline aio/content/guide/deployment.md]: Nit: Ideally there should be a space after `:`:

```diff
-"browserTarget":"myAngularApp:serve:es5"
+"browserTarget": "myAngularApp:serve:es5"
```

**Review by gkalpak — APPROVED** [review]
👌

**Review by kapunahelewong — APPROVED** [review]

File: aio/content/guide/deployment.md [hunk 13]
```diff
@@ -651,24 +651,24 @@ Create an [ES5 serve configuration](guide/deployment#configuring-serve-for-es5)
...
-      "devServerTarget": "app:serve:es5"
+      "devServerTarget": "<app-name>:serve:es5"
     }
   }
 },
 
 </code-example>
 
-You can then run the e2e's with this configuration
+You can then run the `ng e2e` command with this configuration. Make sure to replace `<app-name>` (in `"<app-name>:serve:es5"`) with the actual name of the app, as it appears under `projects` in `angular.json`. For example, if your app name is `myAngularApp` the config will become `"browserTarget": "myAngularApp:serve:es5"`.
```
Comment (gkalpak) [inline aio/content/guide/deployment.md]: I just realized this should be `devServerTarget`, not `browserTarget`:

```diff
-"browserTarget": "myAngularApp:serve:es5"
+"devServerTarget": "myAngularApp:serve:es5"
```

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: aio/content/guide/deployment.md (hunk 13 above)
Comment (santoshyadav198613) [reply aio/content/guide/deployment.md]: Done.

**Review by gkalpak — APPROVED** [review]

**Comment by gkalpak** [comment]
@santoshyadav198613, could you rebase on latest master to get CI green.

**Comment by santoshyadav198613** [comment]
> @santoshyadav198613, could you rebase on latest master to get CI green.

Done.

**Comment by gkalpak** [comment]
**merge-assistance:** Global approval for this docs-only change.
